asab/api/discovery.py

```diff
@@ -30,15 +30,22 @@ def __init__(self, app, zkc, service_name="asab.DiscoveryService") -> None:
 
 		self.App.PubSub.subscribe("Application.tick/300!", self._on_tick)
 		self.App.PubSub.subscribe("ZooKeeperContainer.state/CONNECTED!", self._on_zk_ready)
+		self.App.PubSub.subscribe("Discover.Rescan!", self._on_rescan)
 
 
 	def _on_tick(self, msg):
-		self._update_cache(msg)
+		self._update_cache()
 
+	async def _on_rescan(self, msg):
+		# async method to provide threadsafe callback
+		self._update_cache()
```

Comment:
Be careful with this - it is a method that is / should be called on the non-main thread - we should not introduce this kind of methods.